experiment/migrate_testgrid_tabs.py

```diff
@@ -38,70 +38,121 @@
+    if "presubmits" in prow_config:
+        for _, jobs in prow_config["presubmits"].items():
             for job in jobs:
-                if prowJobName == job["name"]:
-                    job = patch_prow_job(job, dashTab, dashboardName)
+                if prow_job_name == job["name"]:
+                    job = patch_prow_job_with_tab(job, dash_tab, dashboard_name)
 
     # postsubmits -> <any repository> -> [{name: prowjob}]
-    if "postsubmits" in prowConfig:
-        for _, jobs in prowConfig["postsubmits"].items():
+    if "postsubmits" in prow_config:
+        for _, jobs in prow_config["postsubmits"].items():
             for job in jobs:
-                if prowJobName == job["name"]:
-                    job = patch_prow_job(job, dashTab, dashboardName)
+                if prow_job_name == job["name"]:
+                    job = patch_prow_job_with_tab(job, dash_tab, dashboard_name)
 
     # periodics -> [{name: prowjob}]
-    if "periodics" in prowConfig:
-        for job in prowConfig["periodics"]:
-            if prowJobName == job["name"]:
-                job = patch_prow_job(job, dashTab, dashboardName)
+    if "periodics" in prow_config:
+        for job in prow_config["periodics"]:
+            if prow_job_name == job["name"]:
+                job = patch_prow_job_with_tab(job, dash_tab, dashboard_name)
 
     # Dump ProwConfig to prowJobFile
-    with open(prowJobFileName, "w") as prowJobFile:
-        ruamel.yaml.dump(prowConfig, prowJobFile,
+    with open(prow_job_file_name, "w") as job_fp:
+        ruamel.yaml.dump(prow_config, job_fp,
                          Dumper=ruamel.yaml.RoundTripDumper, width=MAX_WIDTH)
-        prowJobFile.truncate()
+        job_fp.truncate()
 
     # delete tab
-    dashboard["dashboard_tab"].remove(dashTab)
+    dashboard["dashboard_tab"].remove(dash_tab)
+
+
+def move_group(config, group, prow_dir):
+    """Moves a given test group to the matching prow job in prow_dir, if possible"""
+    prow_job_name = group["name"]
+    prow_job_file_name = find_prow_job(prow_job_name, prow_dir)
+    if prow_job_file_name == "":
+        return
+
+    # Matching file found; patch and delete
+    print("Patching group {0} in {1}".format(prow_job_name, prow_job_file_name))
+
+    with open(prow_job_file_name, "r") as job_fp:
+        prow_config = ruamel.yaml.load(job_fp,
+                                       Loader=ruamel.yaml.RoundTripLoader,
+                                       preserve_quotes=True)
+
+    # For each presubmit, postsubmit, and periodic
+    # presubmits -> <any repository> -> [{name: prowjob}]
+    # An annotation must be forced, or else the testgroup will not be generated
+    if "presubmits" in prow_config:
+        for _, jobs in prow_config["presubmits"].items():
+            for job in jobs:
+                if prow_job_name == job["name"]:
+                    job = patch_prow_job_with_group(job, group, force_annotation=True)
```

Comment:
a lot of the iteration through prowjobs is copy-pasted between this and `move_tab`, could it be de-duped?